vhdbuilder/scripts/linux/tool_installs.sh

```diff
@@ -45,4 +45,34 @@ disableSystemdIptables() {
 
 enableCgroupV2forAzureLinux() {
     sed -i 's/systemd.legacy_systemd_cgroup_controller=yes systemd.unified_cgroup_hierarchy=0//g' /boot/systemd.cfg
+}
+
+# download and setup azcopy to use to download private packages with MSI auth
+getAzCopyCurrentPath() {
+  if [[ -f ./azcopy ]]; then
+    echo "./azcopy already exists"
+  else
+    echo "get azcopy at \"${PWD}\"...start"
+    # Download and extract
+    azcopydownloadurl="https://azcopyvnext.azureedge.net/releases/release-10.22.1-20231220/azcopy_linux_amd64_10.22.1.tar.gz"
+    azcopysha256="7549424d56ab2d8b4033c84c2a9bb167dc2dcbb23998acd7fffb37bc1a71a267"
+    if [[ $(isARM64) == 1 ]]; then
+      azcopydownloadurl="https://azcopyvnext.azureedge.net/releases/release-10.22.1-20231220/azcopy_linux_arm64_10.22.1.tar.gz"
+      azcopysha256="4db9a4b48abc7775f1a5d6d928afc42361dcc57bbfcde23ac82e4c419a0dc8fc"
+    fi
+
+    downloadedpkg="downloadazcopy"
+    pkgprefix="azcopy_linux_"
+
+    wget "$azcopydownloadurl" -O "$downloadedpkg" &&
```

Comment:
we should be exiting on a new exit code here if the azcopy download fails, you can check how we do it for bpftrace above

I'd also wrap the wget in a `retrycmd_if_failure`